src/segmented_cache_helpers.erl

```diff
@@ -268,31 +300,33 @@ assert_parameters(Opts) when is_map(Opts) ->
                {hours, H} -> timer:hours(H);
                T when is_integer(T) -> timer:minutes(T)
            end,
+    true = is_integer(N) andalso N > 0,
+    true = (EntriesLimit =:= infinity) orelse (is_integer(EntriesLimit) andalso N > 0),
     true = (TTL =:= infinity) orelse (is_integer(TTL) andalso N > 0),
-    Strategy = maps:get(strategy, Opts, fifo),
     true = (Strategy =:= fifo) orelse (Strategy =:= lru),
-    MergerFun = maps:get(merger_fun, Opts, fun segmented_cache_callbacks:default_merger_fun/2),
     true = is_function(MergerFun, 2),
-    Scope = maps:get(scope, Opts, pg),
     true = (undefined =/= whereis(Scope)),
-    {Scope, N, TTL, Strategy, MergerFun}.
+    Opts#{ttl := TTL}.
+
+defaults() ->
+    #{scope => pg,
+      strategy => fifo,
+      entries_limit => infinity,
+      segment_num => 3,
+      ttl => {hours, 8},
+      merger_fun => fun segmented_cache_callbacks:default_merger_fun/2}.
 
--ifdef(OTP_RELEASE).
-  -if(?OTP_RELEASE >= 25).
-    ets_settings() ->
-        [set, public,
-         {read_concurrency, true},
-         {write_concurrency, auto},
-         {decentralized_counters, true}].
-  -elif(?OTP_RELEASE >= 21).
-    ets_settings() ->
-        [set, public,
-         {read_concurrency, true},
-         {write_concurrency, true},
-         {decentralized_counters, true}].
-  -endif.
--else.
-ets_settings() ->
+-if(?OTP_RELEASE >= 25).
+ets_settings(#{entries_limit := infinity}) ->
+    [set, public,
+     {read_concurrency, true},
+     {write_concurrency, auto}];
+ets_settings(#{entries_limit := _}) ->
+    [set, public,
+     {read_concurrency, true},
+     {write_concurrency, true}].
+-elif(?OTP_RELEASE >= 21).
```

Comment:
Why is there `elif`, if in rebar.config, the minimum_otp_vsn is set to 21? Shouldn't it be enough to use `-else`? Or is it added to increase readability?

Reply:
No real reason, this just came from the previous ifdef.